From 410060f91198f1392d0810544411a53f3baa2774 Mon Sep 17 00:00:00 2001
From: Dan Prince <dprince@redhat.com>
Date: Thu, 3 May 2012 10:06:46 -0400
Subject: [PATCH] Use fake_libvirt_utils for libvirt console tests.

Updates test_libvirt.py so that it users fake_libvirt_utils.

This resolves an issue where the previous tests required
sudo to properly execute.

Fixes LP Bug #992805.

Change-Id: I9354b8facf82fb861dc2fe6da5fca102d8a33fbf
(cherry picked from commit 82319bf69745f7a20d0f6273821745302ab711e6)
---
 nova/tests/fake_libvirt_utils.py |    7 ++++++-
 nova/tests/test_libvirt.py       |   11 +++++++----
 2 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/nova/tests/fake_libvirt_utils.py b/nova/tests/fake_libvirt_utils.py
index bdd24e4..f1d1cb7 100644
--- a/nova/tests/fake_libvirt_utils.py
+++ b/nova/tests/fake_libvirt_utils.py
@@ -14,6 +14,7 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
+import os
 import StringIO
 
 files = {}
@@ -89,7 +90,11 @@ def file_open(path, mode=None):
 
 
 def load_file(path):
-    return ''
+    if os.path.exists(path):
+        with open(path, 'r+') as fp:
+            return fp.read()
+    else:
+        return ''
 
 
 def file_delete(path):
diff --git a/nova/tests/test_libvirt.py b/nova/tests/test_libvirt.py
index af30051..afe4630 100644
--- a/nova/tests/test_libvirt.py
+++ b/nova/tests/test_libvirt.py
@@ -1329,7 +1329,6 @@ class LibvirtConnTestCase(test.TestCase):
             shutil.rmtree(os.path.join(FLAGS.instances_path,
                                        FLAGS.base_dir_name))
 
-    @test.skip_if(missing_libvirt(), "Test requires libvirt")
     def test_get_console_output_file(self):
 
         with utils.tempdir() as tmpdir:
@@ -1364,13 +1363,12 @@ class LibvirtConnTestCase(test.TestCase):
 
             self.create_fake_libvirt_mock()
             connection.LibvirtConnection._conn.lookupByName = fake_lookup
-            connection.libvirt_utils = libvirt_utils
+            connection.libvirt_utils = fake_libvirt_utils
 
             conn = connection.LibvirtConnection(False)
             output = conn.get_console_output(instance)
             self.assertEquals("foo", output)
 
-    @test.skip_if(missing_libvirt(), "Test requires libvirt")
     def test_get_console_output_pty(self):
 
         with utils.tempdir() as tmpdir:
@@ -1403,9 +1401,14 @@ class LibvirtConnTestCase(test.TestCase):
             def fake_lookup(id):
                 return FakeVirtDomain(fake_dom_xml)
 
+            def _fake_flush(self, fake_pty):
+                with open(fake_pty, 'r+') as fp:
+                    return fp.read()
+
             self.create_fake_libvirt_mock()
             connection.LibvirtConnection._conn.lookupByName = fake_lookup
-            connection.libvirt_utils = libvirt_utils
+            connection.LibvirtConnection._flush_libvirt_console = _fake_flush
+            connection.libvirt_utils = fake_libvirt_utils
 
             conn = connection.LibvirtConnection(False)
             output = conn.get_console_output(instance)
